testsuite/forge/src/backend/local/cargo.rs

```diff
@@ -162,19 +170,20 @@ pub fn git_merge_base<R: AsRef<str>>(rev: R) -> Result<String> {
 }
 
 pub fn cargo_build_common_args() -> Vec<&'static str> {
-    let use_release = use_release();
-    let consensus_only = build_consensus_only_node();
-    let network_perf_test = build_network_perf_test();
-    let mut args = vec!["build", "--features=failpoints,indexer"];
-    if consensus_only {
+    let mut args = if build_aptos_node_without_indexer() {
+        vec!["build", "--features=failpoints"]
+    } else {
+        vec!["build", "--features=failpoints,indexer"]
+    };
+    if build_consensus_only_node() {
         args.push("--features=consensus-only-perf-test");
     }
-    if network_perf_test {
```

Comment:
@JoshLind I rearranged a thing you just did in merge cleanup, wanted to make sure you still think this looks okay. Note the comments on each build option at the top of this file.